packages/esm-patient-vitals-app/src/vitals/vitals-overview.component.tsx

```diff
@@ -26,84 +25,67 @@ const VitalsOverview: React.FC<VitalsOverviewProps> = ({ patientUuid, showAddVit
   const { t } = useTranslation();
   const displayText = t('vitalSigns', 'Vital signs');
   const headerTitle = t('vitals', 'Vitals');
-
-  const config = useConfig();
-  const { conceptsUnits } = useVitalsSignsConceptMetaData();
-  const [bloodPressureUnit, , temperatureUnit, , , pulseUnit, oxygenSaturationUnit, , respiratoryRateUnit] =
-    conceptsUnits;
-  const [vitals, setVitals] = React.useState<Array<PatientVitals>>(null);
   const [chartView, setChartView] = React.useState<boolean>();
-  const [error, setError] = React.useState(null);
+
+  const { data: vitals, isError, isLoading, isValidating } = useVitals(patientUuid);
+  const { data: conceptData } = useVitalsConceptMetadata();
+  const conceptUnits = conceptData ? conceptData.conceptUnits : null;
+
+  const launchVitalsBiometricsForm = React.useCallback(
+    () => attach('patient-chart-workspace-slot', patientVitalsBiometricsFormWorkspace),
+    [],
+  );
 
   const tableHeaders = [
     { key: 'date', header: 'Date and time', isSortable: true },
     {
       key: 'bloodPressure',
-      header: withUnit('BP', bloodPressureUnit),
+      header: withUnit('BP', conceptUnits ? conceptUnits[0] : ''),
```

Comment:
```suggestion
      header: withUnit('BP', conceptUnits[0] ?? ''),
```

Right? Also I wonder if `withUnit` should just treat `undefined` like `''`.